language/stdlib/modules/LibraVMConfig.move

```diff
@@ -3,15 +3,30 @@ address 0x1 {
 module LibraVMConfig {
     use 0x1::LibraConfig;
     use 0x1::Roles;
+    use 0x1::Vector;
+
+    const SCRIPT_HASH_LENGTH: u64 = 32;
+    const ESCRIPT_HASH_LENGTH_NOT_MATCH: u64 = 0;
 
     // The struct to hold all config data needed to operate the LibraVM.
     // * publishing_option: Defines Scripts/Modules that are allowed to execute in the current configruation.
     // * gas_schedule: Cost of running the VM.
     struct LibraVMConfig {
-        publishing_option: vector<u8>,
+        publishing_option: PublishingOption,
         gas_schedule: GasSchedule,
     }
 
+    struct PublishingOption {
+        // Only script hashes in the following list can be executed by the network.
+        script_allow_list: vector<vector<u8>>,
+        // VM can execute arbitrary script provided by the user if this flag is set to true.
+        is_open_script: bool,
+        // Transactions with sender in this list won't abide to the allow list rules above.
+        sender_allow_list: vector<address>,
```

Comment:
I think this probably should be a role-based check in the transaction prologue instead. i.e.: 
- kill this field
- check if the `&signer` is LR in the prologue, do allowlist check if not

Reply:
I don't hink that's doable at the moment. The reason is that the logic should be:
If sender in list, we don't perform the hash allow list check.

Th logic should live entirely in either rust land or in move land. If we would like to do it in the prologue, we would need to pass other info like if this txn is trying to publish a module/script hash to the prologue